src/bun.js/api/bun/ms.zig

```diff
@@ -0,0 +1,308 @@
+/// Parse a time string like "2d", "1.5h", "5m" to milliseconds
+pub fn parse(input: []const u8) ?f64 {
+    if (input.len == 0 or input.len > 100) return null;
+
+    var i: usize = 0;
+
+    next: switch (input[i]) {
+        '-', '.', '0'...'9' => {
+            i += 1;
+            if (i < input.len) continue :next input[i];
+            break :next;
+        },
+        ' ', 'a'...'z', 'A'...'Z' => break :next,
+        else => return null,
+    }
+
+    const value = std.fmt.parseFloat(f64, input[0..i]) catch return null;
+
+    const unit = strings.trimLeadingChar(input[i..], ' ');
+    if (unit.len == 0) return value;
+
+    if (MultiplierMap.getASCIIICaseInsensitive(unit)) |m| {
+        return value * m;
+    }
+
+    return null;
```

Comment:
See for example, what the possible exit cases of your function are:

- Ran out of memory
- Couldn't parse the integral part.
- No unit given
- Couldn't parse the unit

Yet you lost so much information in having returned `null`. This is not appropriate use of `null`.

Reply:
You don't know what callers might want. Good API design involves, at least partly, predicting the various uses someone may have.

I'm thinking that this library could be used within internal Bun native code, and that may want to care about all the possible various errors.